src/umpire/strategy/DynamicPool.cpp

```diff
@@ -277,6 +277,8 @@ void DynamicPool::mergeFreeBlocks()
 {
   using PointerMap = std::map<Pointer, SizeTuple>;
 
+  if (m_free_map.size() < 2) return;
```

Comment:
This is unrelated, but it's good to return here rather than build a map unnecessarily.

Reply:
This looks fine, although I would be inclined to move this check above the `using Pointermap` declaration on line 278.  I realize it won't make any runtime difference and it is a cosmetic/style thing.  So, if you want to keep the line here, that is fine too.